src/shared/directives/conditional-show/domain/conditional-show.model.ts

```diff
@@ -10,24 +10,24 @@ export class ShowCondition {
   constructor(public condition: string) {
   }
 
-  match(fields): boolean {
+  match(fields, path?: string): boolean {
     if (!this.condition) {
       return true;
     }
-    return this.matchAndConditions(fields, this.condition);
+    return this.matchAndConditions(fields, this.condition, path);
   }
 
-  private matchAndConditions(fields: any, condition: string): boolean {
+  private matchAndConditions(fields: any, condition: string, path?: string): boolean {
     let andConditions = condition.split(ShowCondition.AND_CONDITION_REGEXP);
-    return andConditions.every(andCondition => this.matchEqualityCondition(fields, andCondition));
+    return andConditions.every(andCondition => this.matchEqualityCondition(fields, andCondition, path));
   }
 
-  private matchEqualityCondition(fields: any, condition: string): boolean {
+  private matchEqualityCondition(fields: any, condition: string, path?: string): boolean {
```

Comment:
This method is a bit bulky. Is it possible to break it down a bit?

Reply:
I agree with @andrewfolga, the complexity of this class has increased a lot. We need refactoring and further simplification. It's a bit difficult to follow